jupyter-extension/jupyterlab_pachyderm/pps_client.py

```diff
@@ -17,14 +17,49 @@
 from .log import get_logger
 
 METADATA_KEY = 'pachyderm_pps'
-
+PIPELINE_TYPE_SERVICE = 3
 
 class PPSClient:
     """Client interface for the PPS extension backend."""
 
     def __init__(self):
         self.nbconvert = PythonExporter()
 
+    async def retrieve(self, name, project_name):
+        """Retrieves details about the specified pipeline.
+
+        Args:
+            name: The name of the pipeline
+            project_name: The name of the project
+        """
+        get_logger().debug(f"Retrieving {project_name}/{name}")
+        client = self._client()
+
+        try:
+            for p in client.inspect_pipeline(pipeline_name=name, project_name=project_name, details=True):
+                # This is an incomplete serialization, as
```

Comment:
A couple of thoughts here:
- It might make sense to proxy through the whole object for future expansion (at the cost of some extra bandwidth)
- We might want to match the format returned by the `inspect_pipeline` rpc. That way it's the same all the way through from backend to frontend.